sdk/core/azure-core/azure/core/credentials_async.py

```diff
@@ -0,0 +1,23 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+    from typing_extensions import Protocol
+    from .credentials import AccessToken
+
+    class AsyncTokenCredential(Protocol):
+        async def get_token(self, *scopes: str, **kwargs: Any) -> AccessToken:
+            pass
+
+        async def close(self) -> None:
+            pass
+
+        async def __aenter__(self):
+            pass
+
+        async def __aexit__(self, *args) -> None:
```

Comment:
I would suggest to keep the exact definition in the the Python ABC, for future better typing
async def __aexit__(self, exc_type, exc_value, traceback):
https://github.com/python/cpython/blob/39a5c889d30d03a88102e56f03ee0c95db198fb3/Lib/contextlib.py#L44